tests/FSharp.Compiler.ComponentTests/Conformance/Types/RecordTypes/RecordTypes.fs

```diff
@@ -591,4 +591,16 @@ module RecordTypes =
             (Error 668, Line 5, Col 9, Line 5, Col 10, "The field 'A' appears multiple times in this record expression or pattern")
             (Error 668, Line 5, Col 24, Line 5, Col 25, "The field 'A' appears multiple times in this record expression or pattern")
             (Error 668, Line 5, Col 31, Line 5, Col 32, "The field 'B' appears multiple times in this record expression or pattern")
-        ]
\ No newline at end of file
+        ]
+
+    [<Fact>]
+    let ``Cyclic reference check works for recursive reference with a lifted generic argument`` () =
```

Comment:
Should the error also work when defining such type in sig files ?

Reply:
Good call. If this same logic is run for signature files, then yes.

Reply:
fdbaa66